django_lti_tool_provider/views.py

```diff
@@ -45,10 +46,19 @@ def post(self, request, *args, **kwargs):
         return self.process_request(request)
 
     def process_request(self, request):
+        if request.user.is_authenticated():
+            try:
+                lti_parameters = self._get_lti_parameters_from_request(request)
+                if not self._right_user(request.user, lti_parameters):
+                    _logger.debug(u"Logging out user %s in favor of new LTI session.", request.user.username)
+                    logout(request)
+            except (oauth2.Error, AttributeError):
+                # Not a new visit, or better to keep existing auth.
+                pass
         if not request.user.is_authenticated():
```

Comment:
@smarnach No, because the condition may change by the end of the first if statement, since a user may be logged out. I'm thinking my decision to catch those exceptions and pass may be the incorrect one, though. I think I should log the user out and then let that exception be raised again in the unloggedin state if it still applies. Currently things aren't working right in the deployment and I wonder if that's why.